### ISSUE TITLE
Slimbox OSC layout with volume control area

### ISSUE THREAD

**Issue by JorgeR81** [mpv-player/mpv#8170]
Hi,

I'm looking for a way to control volume and playback scrolling, on the Slimbox OSC layout, just with the mouse wheel, without using the keyboard. 
I already asked a question about this: https://github.com/mpv-player/mpv/issues/8168

But I've thought of a more elegant solution, and the issue is now closed, so I created a new one. I hope that's OK.

What about a new OSC layout with the "slimbox" and volume control area elements.
It's basically the "box" layout, without the elements I don't need.
Just placing the volume control area on the right side of the slimbox.
It could be a new "layout" or a "configurable option", available just for the slimbox layout.

Would that be easy to implement?

I could look into that, but I don't have much coding experience.

I think enough people could be interested in this feature, so that it justifies the MPV team taking time to implement it.
At least keep it open as a low priority feature for future releases. Other developers and users could see it and take interest in the idea.

The "box" and "bottombar" are nice, and they are well organized. The bottombar has a very modern look to it. 
But they have too much information I don't need to see every time I use the seekbar. They are a bit distracting.

For me the slimbox is perfection (see picture attached).

Thanks
Jorge Rodrigues

![Spirited Away with slimbox](https://user-images.githubusercontent.com/72734038/95804102-af28aa80-0cf9-11eb-8673-3a878a68b93a.jpg)

**Comment by shabahengam** [comment]
you can use another OSC like uosc -> https://github.com/darsain/uosc
it has a volume bar on the right and you can use mouse wheel to change the volume.(see the screenshot in uosc repo)

**Comment by JorgeR81** [comment]
Hi,

I've found this fast forward user script, and it just became my favorite way to navigate videos. 
https://github.com/jgreco/mpv-scripts/blob/master/fastforward.lua

Much better than seeking with the mouse wheel. It feels really smooth, really natural (don't know how to explain it better...).

So I guess I will put volume control on the mouse wheel, and use Ctrl+mouse wheel for seeking (when I need to move faster).

I downloaded MPV 5 days ago, just to try it, and now it's quickly becoming my favorite software!

I still think the volume control area would be a good addition to the slimbox (as an option).
So I will leave this issue open, and let the MPV team decide if they want to close it.

Thanks

**Comment by Akemi** [comment]
i think we don't want to change the OSC any more. it's already quite complicated and probably need a rewrite.

**Comment by JorgeR81** [comment]
It's OK, thanks for taking a look in to this.

Since then, I've learned how to modify the OSC myself. 
I've disabled the OSC, in the mpv.conf  ("no-osc" ).
And now I'm using a modified osc.lua script, with all the changes I need.
https://github.com/mpv-player/mpv/blob/master/player/lua/osc.lua

I'm hoping that, if there's a rewrite of the OSC, you can maintain compatibility, so that people can still use their own custom osc.lua scripts, in the new mpv versions.

**Comment by JorgeR81** [comment]
PS: I will check the "Zones" script. It sounds interesting ...

I'm doing something similar in my custom OSC. I've added extra functions to the osc_init() function and now different areas of the OSC, give me different mouse functions, depending on where the mouse pointer is. 
Eg., the mouse wheel can be used to:
- control volume, when the pointer it's on the volume button.
- control seek, when the pointer it's in the seek bar
- control playback speed when the pointer it's in the total time display area
- control frame steps when the pointer it's in the current time display area

These areas are small, but they work right up to the edge of the display. So It can be very fast to use.
Eg. I've relocated the volume button to the right-bottom corner, so I only have to move the mouse to this general area in order to place it the right position, before using the mouse wheel.
